projects/igniteui-angular/src/lib/icon/icon.component.ts

```diff
@@ -120,10 +120,15 @@ export class IgxIconComponent implements OnInit {
      */
     public el: ElementRef;
 
-    constructor(private _el: ElementRef, private iconService: IgxIconService) {
+    constructor(
+            private _el: ElementRef,
+            private iconService: IgxIconService,
+            private ref: ChangeDetectorRef) {
         this.el = _el;
         this.font = this.iconService.defaultFontSet;
         this.iconService.registerFontSetAlias('material', 'material-icons');
+        this.iconService.iconLoaded.pipe(first(e => e.name === this.iconName && e.fontSet === this.font))
```

Comment:
There's a chance this subscription won't be fulfilled if the svg fails to load or something. Or maybe the icon doesn't have name set even. Shouldn't we make sure the subscription is disposed either way?

Reply:
Good catch!